src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -72,5 +75,89 @@ public static string CreateResponseKey(string requestKey)
 
             return Convert.ToBase64String(hashedBytes);
         }
+
+        public static bool ParseDeflateOptions(ReadOnlySpan<char> extension, WebSocketDeflateOptions options, [NotNullWhen(true)] out string? response)
+        {
+            response = null;
+            var builder = new StringBuilder(WebSocketDeflateConstants.MaxExtensionLength);
+            builder.Append(WebSocketDeflateConstants.Extension);
+
+            while (true)
+            {
+                int end = extension.IndexOf(';');
+                ReadOnlySpan<char> value = (end >= 0 ? extension[..end] : extension).Trim();
+
+                if (value.Length > 0)
+                {
+                    if (value.SequenceEqual(WebSocketDeflateConstants.ClientNoContextTakeover))
+                    {
+                        // REVIEW: If someone specifies true for server options, do we allow client to override this?
```

Comment:
> By including this extension parameter in an
   extension negotiation offer, a client informs the peer server of a
   hint that even if the server doesn't include the
   "client_no_context_takeover" extension parameter in the corresponding
   extension negotiation response to the offer, the client is not going
   to use context takeover.

It doesn't seem to matter if the client 'overrides' the server, the client isn't going to use takeover regardless.